testif.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# This script is invoked by Travis testing and inspects
+# the changeset that's being built.
+#
+# Usage:
+#
+#    testif.sh <label> && something_else
+#
+# The intention is to be able to ONLY test certain aspects
+# when special conditions are satisfied.
+#
+# Notes:
+#    JS tests are expected to be very fast
+#    We shouldn't skip tests when it affects Coverage negatively
+#    We can possibly do tests based on which branch is pushed to
+#
+# https://github.com/learningequality/kolibri/issues/27
+#
+# What changes are analyzed?
+#
+# For this script, we should mainly use Travis' env
+# $TRAVIS_COMMIT_RANGE. The reason is simple: You cannot tell what
+# changes have been implemented since last test, thus if your
+# condtion should fire or not.
+#
+# Definition:
+# TRAVIS_COMMIT_RANGE: The range of commits that were included in
+# the push or pull request.
+#
+# https://docs.travis-ci.com/user/environment-variables/
+
+
+set -e
+
+LABEL="$1"
+
+branches_to_always_test=( "master" "releases/*" )
+
+PR="$TRAVIS_PULL_REQUEST"
+commit="$TRAVIS_COMMIT"
+commits="$TRAVIS_COMMIT_RANGE"
+current_branch="$TRAVIS_BRANCH"
+git_changeset=`git show --name-only --no-notes --oneline $commits`
+
+# Some debug stuff, will be removed
+echo "Current branch: $current_branch"
+echo "Commit: $TRAVIS_COMMIT"
+echo "Commit range: $TRAVIS_COMMIT_RANGE"
+echo "PR: $PR"
+echo "Git change set: $git_changeset"
+
+function match_changes {
+    # Usage: match_changes "match1" "match2"
+    # if branch should always be tested
+    if ! [ "$PR" == "false" ]
+    then
+        for branch in "${branches_to_always_test[@]}"
+        do
+            if echo "$current_branch" | grep -q "$branch"
+                echo "Always testing $branch, so not skipping $LABEL..."
+                return 0
+            fi
+        done
+    fi
+
+    # Loop through all arguments
+    for text_to_match
```

Comment:
Yes! Btw it's a special Bash syntax -> http://stackoverflow.com/a/1987331/405682